### ISSUE TITLE
CORS blocks API requests when the X-FusionAuth-TenantId header is present

### ISSUE THREAD

**Issue by rfg-anthony-lim** [FusionAuth/fusionauth-issues#44]
### Description 
I installed version 1.2.1
https://fusionauth.io/docs/v1/tech/reference/cors
The documentation says that the URL pattern **/api/*** is allowed through the CORS filter.
But when I started integrating my ReactJS front-end application, the console/network tab shows "Cross-Origin Request Blocked". 

### Steps to reproduce
1. Open your browser and keep the developer tools/console tab open.
2. Try to send a POST request to /api/login from the browser, or from your API client and make sure the **Host** header is set and is a different origin domain. 
```javascript
fetch('http://172.16.0.222:9011/api/login', {
	mode: 'cors',
	method: 'POST',
	headers: {
		'X-FusionAuth-TenantId': 'e360a05d-ef17-4336-a82f-b3e69320d036',
		'Content-Type': 'application/json'
	},
	body: JSON.stringify('{"loginId":"fusionauth@mailnesia.com","password":"{{password}}","ipAddress":"127.0.0.1","device":"usadfnjasdnknklrassgdsarag","metaData":{"device":{"description":"Insomnia App","lastAccessedAddress":"127.0.0.1","name":"Insomnia","type":"OTHER"}}}')
})
```
3. Check the console logs. It will show the CORS error and the **OPTIONS** request from the Network tab is failing. 

### Expected behavior
It should not be blocked by CORS. The **OPTIONS** request should succeed, and then the **POST** request will be sent, regardless of the response. 

### Screenshots
![cors-browser](https://user-images.githubusercontent.com/45254265/48888681-6a2dcd80-ee6e-11e8-9324-c93b1ca46b0b.png)

### Platform
(Please complete the following information)
 - Device: Desktop
 - OS: Windows 10 (Client), Ubuntu 18.10 (Server)
 - Firefox Developer Edition 64.0b10 (64-bit)

### Additional context
If there should be more options to be filled in the docker compose config file, please let me know.

**Comment by rfg-anthony-lim** [comment]
## Update:
When I tried sending the request without "X-FusionAuth-TenantId" header, it went through without CORS blocking and I got a response. The response was the following:
```json
{"generalErrors": [{
      "code": "[TenantIdRequired]",
      "message": "A Tenant Id is required to complete this request. To complete this request, you may assign a Tenant to your API key, or add the X-FusionAuth-TenantId HTTP request header with the Tenant Id." }
]}
```
So, the required header "X-FusionAuth-TenantId" is being blocked by the CORS policy. Also noting that this header is not documented. https://fusionauth.io/docs/v1/tech/apis/login

**Comment by robotdan** [comment]
Thanks @rfg-anthony-lim for reporting this issue. 

It does look like we are missing the `X-FusionAuth-TenantId` header in our CORS configuration.

This header will apply to nearly all APIs when using more than one tenant so we have not documented it on each API, instead we have currently documented this header in the Tenant tutorial. 

https://fusionauth.io/docs/v1/tech/tutorials/tenants#make-an-api-call-for-a-specific-tenant

Here are two options to use as a work-around until this issue is resolved:
1. Use an API key scoped to a specific tenant. (described in the above tutorial)
2. Modify the CORS configuration on your local install. This will not persist across upgrades.

For option 2, find the `web.xml` configuration file, my configuration is here `/usr/local/fusionauth/fusionauth-app/web/WEB-INF/web.xml `. 

Edit the the `cors.allowed.headers` section to add the `X-FusionAuth-TenantId` header. See my example below with this value added.

```xml
    <init-param>
      <param-name>cors.allowed.headers</param-name>
      <param-value>Content-Type,X-Requested-With,Accept,Origin,Access-Control-Request-Method,Access-Control-Request-Headers,Authorization,Last-Modified,X-FusionAuth-TenantId</param-value>
    </init-param>
```

Let me know this allows you to work around this issue. Are you currently using the JavaScript FusionAuth client?

**Comment by robotdan** [comment]
Targeting patch release 1.2.2

**Comment by rfg-anthony-lim** [comment]
Thank you, @robotdan for the quick fix. We will test this release soon.

**Comment by robotdan** [comment]
Available in version 1.2.2.
https://fusionauth.io/docs/v1/tech/release-notes/
